force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,49 @@
+trigger OpportunityTrigger on Opportunity (before delete, before update) {
+    if (Trigger.isBefore && Trigger.isDelete) {
+        // Question 6 solution               
+        Set<Id> accIds = new Set<Id>();
+        for (Opportunity opp : Trigger.old) {
+            if (opp.AccountId != null) {
+                accIds.add(opp.AccountId);
+            }
+        }
+        
+        Map<Id, Account> accMap = new Map<Id, Account>([SELECT Id, Industry FROM Account WHERE Id IN :accIds]);
+
+        for (Opportunity opp : Trigger.old) {            
+            if (opp.StageName == 'Closed Won') {
+                Account acc = accMap.get(opp.AccountId);                
+                if (acc != null && acc.Industry == 'Banking') {
+                    opp.addError('Cannot delete closed opportunity for a banking account that is won');
+                }
+            }
+        }
+    }
+
+    if (Trigger.isBefore && Trigger.isUpdate) {
+        // Question 5 solution
+        for (Opportunity opp : Trigger.new) {
+            if (opp.Amount < 5000) {                
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
```

Comment:
good. I would check less than or equal too. My assumption since the error message says that the amount must be greater than